### ISSUE TITLE
a question

### ISSUE THREAD

**Issue by chenruolin2019** [sunyiyou/SAFAS#3]
Hi, thanks for your reply. I want to know how to reproduce the results of the experiment？Such as O_C_I_to_M,  there are some differences with the results in the paper. I'm looking forward to your reply!

Sincerely

**Comment by sunyiyou** [comment]
Hi, would you like to give more details?

**Comment by chenruolin2019** [comment]
Hi, I didn't change the parameter settings in train.py, but after running preposess.py, I observed that the sizes of the preprocessed images are not consistent, and I am not sure if this will affect the final result. Moreover, the influence of environmental factors, such as the version of torch, cannot be ruled out. At present, the hter of the two experiments of O_C_I_to_M/O_M_I_to_C are all above 10, which is somewhat different from the results in the paper. But the result of O_C_M_to_I is acceptable now.

**Comment by sunyiyou** [comment]
Hi, it is common for results to go above 10. If you compare with table 2, we report the average performance which is easy to reproduce. Table 1 is the "best possible" results.

**Comment by chenruolin2019** [comment]
Hi, I know emmm I just want to reproduce the results in Table1 first, so could you provide some help to reproduce it?

**Comment by sunyiyou** [comment]
The "best possible" results are produced under a hyper-parameter search like batch size, learning rate, alpha, feat_weight, etc. The default parameter setting is used for the "convergence performance" for consistency.

**Comment by sunyiyou** [comment]
The "best possible" results are produced under a hyper-parameter search like batch size, learning rate, alpha, feat_weight, etc. The default parameter setting is used for the "convergence performance" for consistency.

**Comment by chenruolin2019** [comment]
Emmm does the “hyper-parameter search” need to be manually adjusted?

**Comment by sunyiyou** [comment]
Not sure what you mean by "manually". The fact is that different protocols have different optimal settings, which we meant by "best possible".

**Comment by sunyiyou** [comment]
(I am quite sure prior baselines also apply different hyperparameters in different protocols, since I can't reproduce their performance by default codebase. And that is the motivation of this paper proposing to compare upon convergence, which is fairer and more meaningful.)

**Comment by chenruolin2019** [comment]
OK, thanks for your reply, I understand what you mean. So if I want to reproduce the results in Table 1, do I need to adjust the hyper-parameters by myself?

**Comment by sunyiyou** [comment]
Yes, normally we run a grid search script to find the best number. 

For your reference, I trace back the log and find out the setting I used for Table 1. 

| protocol | train_rotation |  base_lr |  alpha |  feat_loss_weight |    temperature|
| ----| ---- |---- |---- |---- |---- |
|OCI-M|   FALSE |		0.01	   |  0.99	|	0.5		| 0.1  |
| IMO-C|   TRUE  |	     0.01	    | 0.999  |	    0.1	| 0.2|

**Comment by chenruolin2019** [comment]
Oh, Thank you very much! Could you send me the remaining experimental parameters of O_C_M_to_I/I_C_M_to_O if you could find them? I'm sorry for wasting your time.

**Comment by chenruolin2019** [comment]
Or could you provide me the script for the grid search? I could search for the most suitable hyper-parameters by myself.

**Comment by sunyiyou** [comment]
Sure, here is the parameters for O_C_M_to_I/I_C_M_to_O. I also update the [gridsearch_lgt.py](https://github.com/sunyiyou/SAFAS/blob/master/gridsearch_lgt.py) FYI. 

| protocol | train_rotation |  base_lr |  scale | alpha |  feat_loss_weight |    temperature|
| ----| ---- |---- |---- |---- |---- |---- |
|OCM-I|   TRUE |		0.005   |  0.99	|	2 | 0.5		| 0.1  |
|ICM-O|   TRUE  |	   0.005     | 0.999  |	  0.5 |  0.1	| 0.2|
